packages/api-server/api_server/routes/tasks/tasks.py

```diff
@@ -99,10 +99,10 @@ async def query_task_states(
         None, description="comma separated list of requester names"
     ),
     pickup: Optional[str] = Query(
-        None, description="comma separated list of pickup names"
+        None, description="comma separated list of pickup names", deprecated=True
```

Comment:
why is this marked as deprecated? Same for the destination query below

Reply:
They are being replaced with `label=name=value`, in `main`. Let me see if I can port generic label filtering over (already tested sorting doesn't work due to tortoise bug).

Reply:
managed to port label sorting over, this also fixes an issue with filtering where if you try to filter by pickup and destination you get no result even though there is a task that meets the criteria.